salsa-2022-tests/tests/compare_and_swap.rs

```diff
@@ -0,0 +1,57 @@
+//! Test that a setting a field on a `#[salsa::input]`
+//! overwrites and returns the old value.
+
+use salsa_2022_tests::{HasLogger, Logger};
```

Comment:
Yeah, let's rename this, you're right that the name isn't right. Perhaps "swap-and-replace" or "mutate-in-place"?